core/src/saros/session/internal/SarosSession.java

```diff
@@ -787,19 +788,15 @@ private SarosSession(
       final String id,
       IContainerContext context,
       IPreferenceStore localProperties,
+      JID localUserJID,
       JID host,
       IPreferenceStore hostProperties) {
 
-    connectionService = getComponent(context, XMPPConnectionService.class);
-
     this.sessionID = id;
     this.referencePointMapper = new SharedReferencePointMapper();
     this.activityQueuer = new ActivityQueuer();
     this.containerContext = context;
 
-    // FIXME that should be passed in !
-    JID localUserJID = connectionService.getJID();
-
     assert localUserJID != null;
```

Comment:
I think the assertion can be removed here. Either move it to the SessionManager or let the SessionManager correctly handle null values (which is not likely to happen but can theoretically occur if the network goes down etc.)

Reply:
That would duplicate the null check. I did not really see a reason to change the current check as the setup did not really change. The only difference is that the local JID is requested a bit earlier.

Reply:
I cannot find the null check. As long as I remember we simply assumed that a connection is established. Hence the assertion. Even if it does not halt true all the time.

Reply:
> I cannot find the null check.

This is the "null check". I just meant checking before calling requires to check everywhere where the method is called (i.e. in two different places with this method) while having it inside the method unifies the checks. But that just leads us back to the basic question of who should check the parameters, the caller or the called method.

I can make this an `IllegalArgumentException` if you would prefer.